### PR TITLE
Generate correct route when a segment is preceded by a dot

### PR DESCRIPTION
**Pull request by sheperson**
#58 Solved the issue with `people.:format` but it didn't work for something like `people:.name` or `people/.:name`.

This solves the issue for every situation.

### CODE REVIEW

**Comment by tagliala** [comment]
Thanks.

Please squash commits, fix rubocop offenses (run rake rubocop or just rake) and push force so I can merge

File: test/routing_test.rb [hunk 1]
```diff
@@ -298,6 +298,21 @@ def test_route_with_optional_format
     assert_routing '/people.xml', controller: 'people', action: 'index', format: 'xml', locale: 'en'
   end
 
+  def test_routes_that_include_dot
```
Comment (tagliala) [inline test/routing_test.rb]: could you please rename to `test_routes_with_dot`?

File: test/routing_test.rb [hunk 2]
```diff
@@ -298,6 +298,21 @@ def test_route_with_optional_format
...
+    draw_routes do
+      localized do
+        get 'people/.:name', to: 'people#index'
+      end
+
+      localized do
```
Comment (tagliala) [inline test/routing_test.rb]: could you please also use one localized block?

**Comment by sheperson** [comment]
Done.

**Comment by tagliala** [comment]
I will do the optimization in another commit

**Comment by tagliala** [comment]
@sheperson do you mind to run your test suite against the develop branch?

I did a lot of changes, I hope that I didn't break anything before releasing a new version

**Comment by sheperson** [comment]
@tagliala I ran the tests and everything is fine.

**Comment by tagliala** [comment]
@sheperson thanks for the feedback. I've released a new version of the gem (4.3.0) which includes the rewritten module

**Pull request merged**